examples/flutter_gallery/lib/welcome/home.dart

```diff
@@ -0,0 +1,153 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:flutter/material.dart';
+
+import 'step.dart';
+import 'steps/all.dart';
+
+const Color _kWelcomeBlue = Color(0xFF0175c2);
+const int _kAutoProgressSeconds = 10;
+const int _kAutoProgressTransitionMilliseconds = 520;
+const Color footerColor = Color(0xffffffff);
+const TextStyle footerButtonTextStyle = TextStyle(
+  fontSize: 16.0,
+  fontWeight: FontWeight.bold,
+  color: _kWelcomeBlue,
+);
+const double footerButtonHeight = 60.0;
+
+class Welcome extends StatefulWidget {
+  const Welcome({ Key key, this.onDismissed }) : super(key: key);
+  final VoidCallback onDismissed;
+
+  static const String routeName = '/welcome';
+
+  @override
+  _WelcomeState createState() => _WelcomeState();
+}
+
+class _WelcomeState extends State<Welcome> with TickerProviderStateMixin {
+  List<WelcomeStep> _steps;
+  TabController _tabController;
+  PageController _pageController;
+  Timer _autoProgressTimer;
+
+  @override
+  void initState() {
+    super.initState();
+    _steps = <WelcomeStep>[
+      FlutterWelcomeStep(tickerProvider: this),
```

Comment:
I would just let these be 4 widgets that gets returned to the pageview builders. This way they can keep their State